File: frontend/src/Song.js
export const Song = ({ song, handleCheck }) => {
    return (
        <div className="track">
            <div class="cover-art">
                <img src={song.coverArt} alt="Cover Art"></img>
            </div>
            <div className="track-text">
                <span>{song.name}</span>
                <span>{song.artistName}</span>
            </div>
            <div className="album-text">
                <span>{song.albumName}</span>
            </div>
            <div class="song-image"><img src="img/small-spotify.png" alt="Link to Song on Spotify" href={song.spotifyLink}></img></div>
            <div class="like-button">
                <form>
                    <input
                        type="checkbox"
                        id="like"
                        name="like"
                        value="like"
                        checked={song.checked ?? false}
                        onChange={handleCheck}
                    />
                </form>
            </div>
        </div>
    );
};

File: deprecated/src/Song.py
from xmlrpc.client import boolean


class Song:
    uri: str
    name: str
    preview: str
    artistName: str
    albumName: str
    coverArt: str
    spotifyLink: str

    def __init__(self, item) -> None:
        self.uri = item["uri"]
        self.name = item["name"]
        self.preview = item["preview_url"]
        try:
            self.artistName = item["artists"][0]["name"]
        except:
            self.artistName = item["artistName"]
        self.albumName = item["album"]["name"]
        self.coverArt = item["album"]["images"][0]["url"]
        self.spotifyLink = item["external_urls"]["spotify"]


    def toJsonObj(self):
        return {
            "uri": self.uri,
            "name": self.name,
            "preview": self.preview,
            "artistName": self.artistName,
            "albumName": self.albumName,
            "coverArt": self.coverArt,
            "spotifyLink": self.spotifyLink
        }
